### PR TITLE
Goliath dna infusion

### PR DESCRIPTION
**Pull request by GuillaumePrata**
## About The Pull Request
This is a baseline version of the organs and I intend on polishing them more in the future (Hopefully after other faunas get added to the infuser.)

Now, this PR adds goliaths to the DNA infuser at genetics. It gives 4 organs and a final bonus effect.

1- Goliath eyes: Simple mostly filler organ that gives night vision.
2- Goliath lungs: Allow miners to breath either lavaland or the default air mix. As a side effect they can't breath pure O2 anymore so internals can't be used. Stay away from N2O or use your gas mask properly.
3- Goliath heart: Give miner ash storm protection
4- Goliath brain: Turns one of the miner's arm into a tendril goliath hammer that can be used to mine. Like the mounted chainsaw it cannot be dropped, it has slower atk speed, deals 20 damage by default and a bonus 80 to lavaland fauna, it also acts as a baseball bat against fauna so you can dodge being hit back with good timing.
As a side effect, you can't equip gloves as your hand is a big ass hammer...

The extra effect for having all 4 organs is lava immunity for now, I really want to turn it into something more interesting later.

GAGS organs and bonus coderspriter arm:
goliath sprites
If I have time I might animate the arm later.
## Why It's Good For The Game
This add some useful tools for miners if they opt into asking genetics for help and bother to drag a goliath corpses to it.
The organs can be useful on the station, but they will only really shine at lavaland.

We were brainstorming more things that miners can get from the station on their downtime waiting the cargo shuttle to bring their bought gear, this would be a simple and easy power up for miners that can have some small (ignoring the hammer arm) bonus to miners, but small power ups pile up.

I also wrote a hackMD around these organs, their goals, non goals, future possibilities for fauna organs (goliath and others) etc.
https://hackmd.io/@GuillaumePrata/goliath_infusion_organs

## Changelog
:cl: Guillaume Prata
add: Geneticists figured out how to infuse goliath DNA into humanoids! (Many monkeys were harmed in the process!)
add: Goliath eyes for nightvision, lungs to breath at lavaland safely, heart to protect you from ash storms and the brain which turns one of your arms into a tendril hammer.
add: Tendril hammer: Your arm becomes a giant mass of plate and tendril but it won't fit on gloves anymore. While slow to swing around, you can obliterate fauna/megafauna with it, 20 base dmg + 80 bonus damage to fauna/megafauna with a bonus knockback.
/:cl:

### CODE REVIEW

File: code/game/machinery/dna_infuser/organ_sets/goliath_organs.dm [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+	desc = "goliath DNA infused into what was once some normal lungs."
+
+	icon = 'icons/obj/medical/organs/infuser_organs.dmi'
+	icon_state = "lungs"
+	greyscale_config = /datum/greyscale_config/mutant_organ
+	greyscale_colors = GOLIATH_COLORS
+
+/obj/item/organ/internal/lungs/lavaland/goliath/Initialize(mapload)
+	. = ..()
+	AddElement(/datum/element/noticable_organ, "small tendrils grow on their back.", BODY_ZONE_HEAD)
```
Comment (tralezab) [inline code/game/machinery/dna_infuser/organ_sets/goliath_organs.dm:60]: don't forget to change body zone on this

File: code/game/machinery/dna_infuser/infuser_entries.dm [hunk 2]
```diff
@@ -135,3 +135,25 @@ GLOBAL_LIST_INIT(infuser_entries, prepare_entries())
...
+	name = "Goliath"
+	infuse_mob_name = "Goliath"
+	desc = "The guy who said 'Whoever fights monsters should see to it that in the process he does not become a monster' clearly didn't see what a goliath miner can do!"
+	threshold_desc = "You can walk on lava!"
+	qualities = list(
+		"can breath in lavaland",
+		"immune to ashstorms",
+		"eyes that can see in the dark",
+		"a tendril hand can easily dig through basalt and obliterate hostile fauna, won't be fitting on gloves any time soon tho...",
+	)
```
Comment (tralezab) [inline code/game/machinery/dna_infuser/infuser_entries.dm:149]: might wanna mention the air breathing.

**Review by tralezab — APPROVED** [review]
great addition, full review when done

**Comment by Striders13** [comment]
kinetic crusher fans are left out, fucked up.

**Comment by GuillaumePrata** [comment]
> kinetic crusher fans are left out, fucked up.

And MODsuit fans.

**Comment by Jacquerel** [comment]
There are many beasties on Lavaland, I’m sure a different one could provide a different role if you wanted one which enhanced your equipment.  
This doesn’t seem so much like it’s for PKA OR Crusher users but supplants both, which is good and fun.

**Comment by Shadowflame909** [comment]
Mining - genetics departmental cooperation. Two departments both lacking in communication shall now communicate with each other. Amazing PR that I hope gets merged as soon as its done

**Comment by DaydreamIQ** [comment]
The hammer seems a bit too overtuned considering the 'downside' is just not being able to have gloves. When the Crusher requires two hands and does less damage while requiring you to also shoot the fauna/backstab to get more damage out of it, this seems wildly imbalanced. Plus giving the crew access to free 20 force nodrop armblades with no real downsides is just gonna lead to powergaming out the ass

**Comment by Overuseddart** [comment]
The fauna damage is a bit high though, it out damages the crusher while making it so you dont have to go for a backstab

**Comment by Overuseddart** [comment]
Is there any interaction between the Goliath eyes and the eye of god? Could be fun.

**Comment by GuillaumePrata** [comment]
> The hammer seems a bit too overtuned considering the 'downside' is just not being able to have gloves.

Another downside is that your hand is always stuck being a weapon and you can't use two handed items like the crusher.

The comparison to the crusher is good, the crusher does 20 damage + 50 extra if the target is marked and it can be upgraded further with different perks without needing to interact with any department which makes the upgrades quite consistent overall.

The tendril hand does 20 damage and an extra 80 to mining and bosses with a small knockback, but it takes twice as long to "recharge" an attack so you can do the next one. (I forgot to put the slower atk spd on the PR so that is on me)
It will lose to basically any other weapon on a DPS race and that was the goal, you walk in, slap the monster to show their place and it is out of range to being counterattacked if it is a melee monster.

If punching every fauna becomes an issue balance wise we can always slow the atk speed further to reward using it in a combo with another weapon like the PKA.

**Comment by GuillaumePrata** [comment]
I tried to fix the linter issue three times but it hates me for some reason so I need some help with it (if it is even important...)

**Comment by private-tristan** [comment]
seeing how these give you organs, it'll be intresting to see if miners decide to harvest the organs so that they can get the effects they want (through a med doc implanting the organs) instead of getting all of them. quite interesting

**Comment by Oranggu** [comment]
This is cool

**Comment by gueba14** [comment]
any plans on doing any other megafauna?

**Comment by Overuseddart** [comment]
An interaction between Goliath eyes and Eye of God could be something like, upon applying eye of god effect a square of 4x4 tentacles appears, dealing 80 damage to fauna and stunning humanoids. Could be fun as Goliath eyes would just grant mostly the same effects as the eye.

**Comment by CCC23** [comment]
:(

**Comment by gueba14** [comment]
this is sad, it was a good concept that was going on with the new stuff

**Comment by catrock31** [comment]
man this looks so cool i can't wait for it to be in

**Comment by catrock31** [comment]
i am vibrateing just waiting for it

**Comment by Livrah** [comment]
Genetics is always lacks things to do and interactions, this looks really good.

**Comment by catrock31** [comment]
> Genetics is always lacks things to do and interactions, this looks really good.

yeah and the design doc has more that wants to be done after this one so even more interactions

by the way are you the cool mime on youtube?

File: code/game/machinery/dna_infuser/organ_sets/goliath_organs.dm [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+	desc = "goliath DNA infused into what was once some normal lungs."
+
+	icon = 'icons/obj/medical/organs/infuser_organs.dmi'
+	icon_state = "lungs"
+	greyscale_config = /datum/greyscale_config/mutant_organ
+	greyscale_colors = GOLIATH_COLORS
+
+/obj/item/organ/internal/lungs/lavaland/goliath/Initialize(mapload)
+	. = ..()
+	AddElement(/datum/element/noticable_organ, "small tendrils grow on their back.", BODY_ZONE_CHEST)
```
Comment (tralezab) [inline code/game/machinery/dna_infuser/organ_sets/goliath_organs.dm:60]: You're going to want to reword this, it'll come out wrong.

File: code/game/machinery/dna_infuser/organ_sets/goliath_organs.dm [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+#define GOLIATH_SCLERA_COLOR "#ac0f32"
+#define GOLIATH_PUPIL_COLOR "#FF0000"
+
+#define GOLIATH_COLORS GOLIATH_ORGAN_COLOR + GOLIATH_SCLERA_COLOR + GOLIATH_PUPIL_COLOR
+
+///bonus of the goliath: you can swim through space!
+/datum/status_effect/organ_set_bonus/goliath
+	organs_needed = 4
+	bonus_activate_text = span_notice("goliath DNA is deeply infused with you! You can now endure walking on lava!")
+	bonus_deactivate_text = span_notice("Your DNA is once again mostly yours, and so fades your ability to survive on lava...")
```
Comment (tralezab) [inline code/game/machinery/dna_infuser/organ_sets/goliath_organs.dm:12]: Would prefer a better description for the first part, "Your DNA is once again mostly yours". I set a bad pattern by making them generic, would prefer if they all had a unique DNA loss message for the first half.

File: code/game/machinery/dna_infuser/organ_sets/goliath_organs.dm (hunk 3 above)
Comment (tralezab) [reply code/game/machinery/dna_infuser/organ_sets/goliath_organs.dm:60]: "His small tendrils grow on their back"

File: code/game/machinery/dna_infuser/organ_sets/goliath_organs.dm [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+	icon = 'icons/obj/medical/organs/infuser_organs.dmi'
+	icon_state = "heart"
+	greyscale_config = /datum/greyscale_config/mutant_organ
+	greyscale_colors = GOLIATH_COLORS
+
+	organ_traits = list(TRAIT_ASHSTORM_IMMUNE)
+
+/obj/item/organ/internal/heart/goliath/Initialize(mapload)
+	. = ..()
+	AddElement(/datum/element/noticable_organ, "thick hide plates are growing under the skin", BODY_ZONE_CHEST)
```
Comment (tralezab) [inline code/game/machinery/dna_infuser/organ_sets/goliath_organs.dm:174]: "His thick plates are growing under the skin"

**Review by tralezab — CHANGES_REQUESTED** [review]
small things

**Comment by adamkad1** [comment]
How do I upgrade mah hammerarm tho

**Pull request merged**